File: CS/WindowsApplication1/MyLayoutViewScrollHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Layout;
using DevExpress.XtraGrid.Views.Layout.ViewInfo;
using DevExpress.XtraGrid.Views.Layout.Events;

namespace WindowsApplication1
{
    public class MyLayoutViewScrollHelper
    {
        private LayoutView _SelectedLayoutView;
        public LayoutView SelectedLayoutView
        {
            get { return _SelectedLayoutView; }
            set { _SelectedLayoutView = value; }
        }

        public MyLayoutViewScrollHelper(LayoutView view)
        {
            SelectedLayoutView = view;
            SelectedLayoutView.VisibleRecordIndexChanged += SelectedLayoutView_VisibleRecordIndexChanged;
        }

        private int GetVisibleCardsCount()
        {
            int visibleCardsCount = (SelectedLayoutView.GetViewInfo() as LayoutViewInfo).VisibleCards.Count;
            return visibleCardsCount;
        }
        int GetPageByVisibleIndex(int visibleIndex)
        {
            return visibleIndex / GetVisibleCardsCount();
        }

        bool IsScrollForward(LayoutViewVisibleRecordIndexChangedEventArgs e)
        {
            return GetPageByVisibleIndex(e.VisibleRecordIndex) == GetPageByVisibleIndex(e.PrevVisibleRecordIndex);
        }

        int GetFirstCardIndex(int pageIndex, bool isForward)
        {
            int delta = isForward ? 1 : 0;
            return GetVisibleCardsCount() * (pageIndex + delta);
        }

        bool locked;

        private void ScrollPage(LayoutViewVisibleRecordIndexChangedEventArgs e)
        {
            SelectedLayoutView.VisibleRecordIndex = e.PrevVisibleRecordIndex;
            SelectedLayoutView.VisibleRecordIndex = GetFirstCardIndex(GetPageByVisibleIndex(e.VisibleRecordIndex), IsScrollForward(e));
        }
        void SelectedLayoutView_VisibleRecordIndexChanged(object sender, LayoutViewVisibleRecordIndexChangedEventArgs e)
        {
            if (locked) return;
            locked = true;
            ScrollPage(e);
            locked = false;
        }
    }
}

File: VB/WindowsApplication1/MyLayoutViewScrollHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Layout
Imports DevExpress.XtraGrid.Views.Layout.ViewInfo
Imports DevExpress.XtraGrid.Views.Layout.Events

Namespace WindowsApplication1
	Public Class MyLayoutViewScrollHelper
		Private _SelectedLayoutView As LayoutView
		Public Property SelectedLayoutView() As LayoutView
			Get
				Return _SelectedLayoutView
			End Get
			Set(ByVal value As LayoutView)
				_SelectedLayoutView = value
			End Set
		End Property

		Public Sub New(ByVal view As LayoutView)
			SelectedLayoutView = view
			AddHandler SelectedLayoutView.VisibleRecordIndexChanged, AddressOf SelectedLayoutView_VisibleRecordIndexChanged
		End Sub

		Private Function GetVisibleCardsCount() As Integer
			Dim visibleCardsCount As Integer = (TryCast(SelectedLayoutView.GetViewInfo(), LayoutViewInfo)).VisibleCards.Count
			Return visibleCardsCount
		End Function
		Private Function GetPageByVisibleIndex(ByVal visibleIndex As Integer) As Integer
            Return visibleIndex \ GetVisibleCardsCount()
		End Function

		Private Function IsScrollForward(ByVal e As LayoutViewVisibleRecordIndexChangedEventArgs) As Boolean
			Return GetPageByVisibleIndex(e.VisibleRecordIndex) = GetPageByVisibleIndex(e.PrevVisibleRecordIndex)
		End Function

		Private Function GetFirstCardIndex(ByVal pageIndex As Integer, ByVal isForward As Boolean) As Integer
			Dim delta As Integer
			If isForward Then
				delta = 1
			Else
				delta = 0
			End If
			Return GetVisibleCardsCount() * (pageIndex + delta)
		End Function

		Private locked As Boolean

		Private Sub ScrollPage(ByVal e As LayoutViewVisibleRecordIndexChangedEventArgs)
			SelectedLayoutView.VisibleRecordIndex = e.PrevVisibleRecordIndex
			SelectedLayoutView.VisibleRecordIndex = GetFirstCardIndex(GetPageByVisibleIndex(e.VisibleRecordIndex), IsScrollForward(e))
		End Sub
		Private Sub SelectedLayoutView_VisibleRecordIndexChanged(ByVal sender As Object, ByVal e As LayoutViewVisibleRecordIndexChangedEventArgs)
			If locked Then
				Return
			End If
			locked = True
			ScrollPage(e)
			locked = False
		End Sub
	End Class
End Namespace
